### ISSUE TITLE
Add writable property to _socket 

### ISSUE THREAD

**Issue by henopied** [uWebSockets/uWebSockets#195]
This would help because some ws code im working with uses _socket.writable, you may want to add more properties to _socket in general too.

**Comment by alexhultman** [comment]
I checked your send code and I think you are using this property wrong. writable is not something you want to check, a net.Socket is always writable, it has internal buffering. I cannot see any documentation about writable in Streams or net.Socket other than as being a settable property on creation.

https://nodejs.org/api/net.html#net_new_net_socket_options
https://nodejs.org/api/net.html#net_socket_buffersize

send always succeeds (if the socket is open). Use the callbacks of send to determine if you want to send more data. If you send a bunch of messages you want to wait until enough callbacks got called.

**Comment by lpinca** [comment]
> a net.Socket is always writable

Not correct, a socket is no longer writable after you called `end` on it.

```js
'use strict';

const net = require('net');

const server = net.createServer((socket) => {
  console.log(socket.writable);
  socket.end();
  console.log(socket.writable);
});

server.listen(3000, () => {
  const socket = new net.Socket();

  socket.on('end', () => {
    socket.end();
    server.close();
  });

  socket.connect(3000);
});
```

**Comment by alexhultman** [comment]
Why would anyone check if the websocket.readyState == OPEN only to then also check that the websocket._socket.writable == true? MultiOgar are seeing cases where these two don't match. How is this? Shouldn't the readyState always be tied to the state of the net.Socket? I mean, clearly readyState == OPEN should not possible happen when the actual net.Socket has writable == false?

**Comment by lpinca** [comment]
I honestly have no idea why they check both `websocket.readyState` and `websocket._socket.writable`.

**Comment by alexhultman** [comment]
Exactly, they check both because they do not match which to me sounds like their fault for somehow messing with "internal" variables. Any implementation of _socket.writable would be (readyState === OPEN).

**Comment by lpinca** [comment]
`_socket.writable` is also `true` when `readyState` is `CONNECTING` or `CLOSING` but yes if `readyState` is `OPEN`, `_socket.writable` should be `true`.

**Comment by alexhultman** [comment]
@henopied I can't even get uws working with MultiOgar at all. It just stops sending data after a couple of messages. It works with ws. I added some console.logs in sendPacket and the game server just stops sending altogether after a couple of messages. Something is very strange with that game server, but I also know I played on your server for an hour the other day so yeah, I don't know. I just cant get it to work.
